controllers/subscriptions.go

```diff
@@ -153,3 +135,71 @@ func (r *StorageSystemReconciler) isVendorCsvReady(instance *odfv1alpha1.Storage
 		return err
 	}
 }
+
+// GetSubscriptions returns all required Subscriptions for the given StorageKind
+func GetSubscriptions(k odfv1alpha1.StorageKind) []*operatorv1alpha1.Subscription {
```

Comment:
How about passing the instance itself to the func, It will give us the freedom to even change the logic afterwards without changing args.

Reply:
I try to follow a principle of passing as little information as is strictly required. Changing a function signature is a trivial change with most IDEs, and in particular it will highlight all references to the function in question so the developer remembers to look at it.